hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NSSummaryEndpoint.java

```diff
@@ -0,0 +1,640 @@
+      assert (names.length == 2);
+      long bucketObjectId = getBucketObjectId(names);
+      basicResponse.setNumTotalDir(getTotalDirCount(bucketObjectId));
+      basicResponse.setNumTotalKey(getTotalKeyCount(bucketObjectId));
+      break;
+    case DIRECTORY:
+      // path should exist so we don't need any extra verification/null check
+      long dirObjectId = getDirObjectId(names);
+      basicResponse = new BasicResponse(EntityType.DIRECTORY);
+      basicResponse.setNumTotalDir(getTotalDirCount(dirObjectId));
+      basicResponse.setNumTotalKey(getTotalKeyCount(dirObjectId));
+      break;
+    case KEY:
+      basicResponse = new BasicResponse(EntityType.KEY);
+      break;
+    case UNKNOWN:
+      basicResponse = new BasicResponse(EntityType.UNKNOWN);
+      basicResponse.setStatus(PathStatus.PATH_NOT_FOUND);
+      break;
+    default:
+      break;
+    }
+    return Response.ok(basicResponse).build();
+  }
+
+  /**
+   * DU endpoint to return datasize for subdirectory (bucket for volume).
+   * @param path request path
+   * @return DU response
+   * @throws IOException
+   */
+  @GET
+  @Path("/du")
+  public Response getDiskUsage(@QueryParam("path") String path)
+          throws IOException {
+    String[] names = parseRequestPath(path);
+    EntityType type = getEntityType(names);
+    DUResponse duResponse = new DUResponse();
+    switch (type) {
+    case VOLUME:
+      String volName = names[0];
+      List<OmBucketInfo> buckets = listBucketsUnderVolume(volName);
+      duResponse.setCount(buckets.size());
+
+      // List of DiskUsage data for all buckets
+      List<DUResponse.DiskUsage> bucketDuData = new ArrayList<>();
+      for (OmBucketInfo bucket : buckets) {
+        String bucketName = bucket.getBucketName();
+        long bucketObjectID = bucket.getObjectID();
+        String subpath = omMetadataManager.getBucketKey(volName, bucketName);
+        DUResponse.DiskUsage diskUsage = new DUResponse.DiskUsage();
+        diskUsage.setSubpath(subpath);
+        long dataSize = getTotalSize(bucketObjectID);
+        diskUsage.setSize(dataSize);
+        bucketDuData.add(diskUsage);
+      }
+      duResponse.setDuData(bucketDuData);
+      break;
+    case BUCKET:
+      long bucketObjectId = getBucketObjectId(names);
+      NSSummary bucketNSSummary =
+              reconNamespaceSummaryManager.getNSSummary(bucketObjectId);
+
+      // get object IDs for all its subdirectories
+      Set<Long> bucketSubdirs = bucketNSSummary.getChildDir();
+      duResponse.setCount(bucketSubdirs.size());
+      List<DUResponse.DiskUsage> dirDUData = new ArrayList<>();
+      for (long subdirObjectId: bucketSubdirs) {
+        NSSummary subdirNSSummary = reconNamespaceSummaryManager
+                .getNSSummary(subdirObjectId);
+
+        // get directory's name and generate the next-level subpath.
+        String dirName = subdirNSSummary.getDirName();
+        String subpath = path + OM_KEY_PREFIX + dirName;
```

Comment:
If `path` already has a trailing `OM_KEY_PREFIX`, don't append `OM_KEY_PREFIX` again. Otherwise it ends up with double `OM_KEY_PREFIX` like this:

e.g. `path=/vol1/bucket2/dir3/`

![image](https://user-images.githubusercontent.com/50227127/127050872-3d199bbd-b77b-4dd1-90e0-6e4037cc33e3.png)